omnigibson/envs/env_base.py

```diff
@@ -35,7 +35,7 @@ class Environment(gym.Env, GymObservable, Recreatable):
     Core environment class that handles loading scene, robot(s), and task, following OpenAI Gym interface.
     """
 
-    def __init__(self, configs):
+    def __init__(self, configs, in_vec_env=False):
```

Comment:
@cgokmen what do you think?

Reply:
+1

Reply:
It should not be part of the config - the nice thing about the vec env setup currently is that you can load the same config into a vec env and you're essentially cloning the environment. If you put it in the config and you load that config into a singular environment class that makes no sense, the decision to load env vs vec env happens in code and not in config.

There a bunch of these things (e.g. vec env calls private methods post_step etc.) that I agree are really ugly. This is due to the singular-first nature of the environment class currently. I would like to flip that (e.g. have VectorEnvironment be the base environment class and Environment be a n=1 special case that simply unwraps everything). There are lots of advantages to this because it lets us implement batch-native tasks, rewards, etc. but that will happen in a follow up.

Reply:
TODO: create a ticket